server/leafnode.go

```diff
@@ -212,18 +233,13 @@ func (s *Server) connectToRemoteLeafNode(remote *leafNodeCfg, firstConnect bool)
 	resolver := s.leafNodeOpts.resolver
 	s.mu.Unlock()
 
-	if remote != nil {
-		remote.Lock()
-		loopDelay := remote.loopDelay
-		remote.loopDelay = 0
-		remote.Unlock()
-		if loopDelay > 0 {
-			select {
-			case <-time.After(loopDelay):
-			case <-s.quitCh:
-				return
-			}
+	if loopDelay := remote.getLoopDelay(); loopDelay > 0 {
+		select {
+		case <-time.After(loopDelay):
+		case <-s.quitCh:
+			return
 		}
+		remote.resetLoopDelay()
```

Comment:
Should we rest only after we successfully connect?

Reply:
Once we pass this code, we enter the loop that tries to create tcp connection. Once tcp connection succeeds, we call createLeafNode() and that does not really say if this is going to be a success or not. If there is a loop again, server will set the loop delay again.
Unless you are saying that server should sleep that amount of time between TCP create from the list of URLs, I think code is ok as it is.

Reply:
Ah ok.